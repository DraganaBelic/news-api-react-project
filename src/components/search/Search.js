import React, { useState, useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { TopNewsWrapperStyle } from "../topNews/TopNewsStyle";
import Thumbnail from "../thumbnail/Thumbnail";
import { search } from "../../api/apiUtil";
import { disableCountrySelection } from "../../actions/countryActions";
import { useTranslation } from "react-i18next";
import { SearchDivStyle } from "./SearchStyle";
import LoaderComponent from "../../general/LoaderComponent";

const Search = () => {
  const currentCountry = useSelector((state) => state.country.currentCountry);
  const dispatch = useDispatch();
  const fetchData = useSelector((state) => state.search);
  const response = fetchData.response;
  const [query, setQuery] = useState("");

  const { t } = useTranslation();
  const handleKeyDown = (event) => {
    if (event.key === "Enter") {
      setQuery(event.target.value);
    }
  };
  useEffect(() => {
    dispatch(disableCountrySelection(false));
    if (query) {
      dispatch(search(query));
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [query]);
  let fetchedNews =
    response && response.articles
      ? response.articles.map((tn, i) => {
          return <Thumbnail key={i} article={tn} />;
        })
      : "";
  return (
    <div>
      <SearchDivStyle>
        <h2>
          {t("searchTopNewsFrom")} {t(currentCountry)} {t("byTerm")}:
        </h2>
        <input
          placeholder={t("searchTerm")}
          onKeyDown={handleKeyDown}
          type="search"
        ></input>
        {fetchData && fetchData.loading ? (
          <LoaderComponent />
        ) : (
          <TopNewsWrapperStyle>{fetchedNews}</TopNewsWrapperStyle>
        )}
      </SearchDivStyle>
    </div>
  );
};

export default Search;
