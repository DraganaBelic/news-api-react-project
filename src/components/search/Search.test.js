import React from "react";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import thunk from "redux-thunk";
import { MemoryRouter } from "react-router-dom";
import Search from "./Search";

const middlewares = [thunk];
const initialState = {
  country: {
    currentCountry: "gb",
  },
  search: {
    fetchData: {
      articles: [
        {
          title: "article",
          urlToImage: "url",
          description: "desc",
        },
      ],
    },
  },
};
export const mockedStore = (state = initialState) => {
  const mockStore = configureStore(middlewares);
  return mockStore(state);
};
jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (s) => s }),
}));
describe("Test Search component", () => {
  const store = mockedStore();
  const { container } = render(
    <Provider store={store}>
      <MemoryRouter>
        <Search />
      </MemoryRouter>
    </Provider>
  );
  it("should render with given state from Redux store", () => {
    expect(container.firstChild).toMatchSnapshot();
  });
});
