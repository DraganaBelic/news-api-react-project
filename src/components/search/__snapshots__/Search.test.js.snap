// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Search component should render with given state from Redux store 1`] = `
<div>
  <div
    class="SearchStyle__SearchDivStyle-sc-1sh7sb9-0 cekAVM"
  >
    <h2>
      searchTopNewsFrom
       
      gb
       
      byTerm
      :
    </h2>
    <input
      placeholder="searchTerm"
      type="search"
    />
    <div
      class="TopNewsStyle__TopNewsWrapperStyle-sc-71icot-0 gbkRyb"
    />
  </div>
</div>
`;
